metrics/report/grabdata.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Run a set of the metrics tests to gather data to be used with the report
+# generator. The general ideal is to have the tests configured to generate
+# useful, meaninful and repeatable (stable, with minimised variance) results.
+# If the tests have to be run more or longer to achieve that, then generally
+# that is fine - this test is not intended to be quick, it is intended to
+# be repeatable.
+
+# Note - no 'set -e' in this file - if one of the metrics tests fails
+# then we wish to continue to try the rest.
+# Finally at the end, in some situations, we explicitly exit with a
+# failure code if necessary.
+
+SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_DIR}/../lib/common.bash"
+RESULTS_DIR=${SCRIPT_DIR}/../results
+
+# Set up the initial state
+init() {
+	metrics_onetime_init
+}
+
+# Execute metrics scripts
+run() {
+	pushd "$SCRIPT_DIR/.."
+
+	# If KSM is available on this platform, let's run any tests that are
+	# affected by having KSM on/orr first, and then turn it off for the
+	# rest of the tests, as KSM may introduce some extra noise in the
+	# results by stealing CPU time for instance.
+	if [[ -f ${KSM_ENABLE_FILE} ]]; then
+		save_ksm_settings
+		trap restore_ksm_settings EXIT QUIT KILL
+		set_ksm_aggressive
+
+		# Run the memory footprint test - the main test that
+		# KSM affects.
+		bash density/docker_memory_usage.sh 20 300 auto
```

Comment:
should `200` `300` and `auto` be defined as variables?

Reply:
Maybe, but they are only used once, and I'm not sure it would add clarity. What might help is if I add a comment above the call explaining why those values - would that suffice?

Reply:
@grahamwhaley yes please just put what those values mean, I think if someone wants to do a change in the scripts is kind of difficult to understand at first glance, to really understand it you will have to be familiar with the tests thank you